app/services/listing_service/api/shapes.rb

```diff
@@ -42,10 +56,19 @@ def update(community_id:, listing_shape_id: nil, opts:)
       }
     end
 
-    def delete(community_id:, listing_shape_id:)
+    def delete(community_id:, listing_shape_id: nil, name: nil)
+      if listing_shape_id.present? && name.present?
+        return Result::Error.new("Can not have both listing shape id (#{listing_shape_id}) and name (#{name}) present.")
+      end
+
+      unless listing_shape_id.present? || name.present?
+        return Result::Error.new("Must have either id or name present.")
+      end
```

Comment:
These two checks could be in private method because they are used from two places.